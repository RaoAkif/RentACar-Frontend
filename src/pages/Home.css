.latest-models-main {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.carousel-container {
  width: 100%;
  margin: 0;
  padding: 0;
}

.carousel-container a {
  all: unset;
  cursor: pointer;
}

.react-multiple-carousel__arrow {
  display: none;
}

.react-multiple-carousel__arrow--left {
  border-radius: 0 4rem 4rem 0;
}

.react-multiple-carousel__arrow--right {
  border-radius: 4rem 0 0 4rem;
}

.react-multiple-carousel__arrow::before {
  height: 2rem;
  width: 2rem;
  margin: 1rem 2rem 2rem 0;
  cursor: pointer;
  display: block;
}

@media only screen and (min-width: 480px) {
  .latest-models-main {
    padding-left: 20rem;
  }

  .carousel-container {
    width: 100%;
  }

  .react-multiple-carousel__arrow {
    margin-left: -6rem;
    margin-right: -6rem;
    width: 5.5rem;
    height: 4rem;
    font-size: 0.5rem;
    display: flex;
    justify-content: flex-end;
    align-items: center;
    color: #fff;
    background-color: var(--green-color);
  }
}
