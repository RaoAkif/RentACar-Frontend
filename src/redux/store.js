import { configureStore } from '@reduxjs/toolkit';
import cakeReducer from './cake/cakeSlice';
import icecreamReducer from './icecream/icecreamSlice';
import userReducer from './user/userSlice';

const store = configureStore({
  reducer: {
    cake: cakeReducer,
    icecream: icecreamReducer,
    user: userReducer,
  },
});

export default store;
