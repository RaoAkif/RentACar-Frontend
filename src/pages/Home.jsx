import React, { useEffect } from 'react';
import Carousel from 'react-multi-carousel';
import { useSelector, useDispatch } from 'react-redux';
import { fetchCars } from '../redux/car/car';
import CarCard from '../components/CarCard';
import 'react-multi-carousel/lib/styles.css';
import './Home.css';

function Home() {
  const dispatch = useDispatch();
  const cars = useSelector((store) => store.car);

  useEffect(() => {
    dispatch(fetchCars());
  }, [dispatch]);

  const responsive = {
    superLargeDesktop: {
      breakpoint: { max: 4000, min: 3000 },
      items: 1,
    },
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 1,
    },
    laptop: {
      breakpoint: { max: 1024, min: 768 },
      items: 1,
    },
    tablet: {
      breakpoint: { max: 768, min: 464 },
      items: 1,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 1,
    },
  };

  const emptyCarsPage = <h2>There are no cars available</h2>;

  const carsPage = (
    <Carousel responsive={responsive} className="carousel-container" infinite>
      {cars.map((car) => (
        <div key={car.id}>
          <a href={`/car_detail/${car.id}`}>
            <CarCard
              key={car.id}
              carName={car.name}
              model={car.model}
              price={car.rent}
              image={car.image}
              city={car.city}
              date={car.date}
            />
          </a>
        </div>
      ))}
    </Carousel>
  );

  if (cars.length === 0) {
    return emptyCarsPage;
  }

  return carsPage;
}

export default Home;
