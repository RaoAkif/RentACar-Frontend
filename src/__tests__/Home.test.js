import React from 'react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import { render } from '@testing-library/react';
import Home from '../pages/Home';

describe('Home page', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
