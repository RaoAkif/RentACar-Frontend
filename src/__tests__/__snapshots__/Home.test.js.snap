// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render correctly 1`] = `
<div>
  <h2>
    There are no cars available
  </h2>
</div>
`;
